libs/api.js

```diff
@@ -97,7 +102,23 @@ class API {
      * @returns {Promise<Object>} a promise that contains the parsed JSON returned from the server
      */
     sessionUpload(access_key, session_id, data) {
-        return this._post(`${access_key}/upload/${session_id}/`, data);
+        let blob = new Blob([data], {type: 'text/plain'});
+        return this._post(`${access_key}/upload/${session_id}/`, blob);
```

Comment:
This isn't a problem introduced in this PR, but it might be better to either have a constant to use for the URL (as the normal upload function does), or have some kind of helper to construct URLs. 

Mostly  so we don't have to hunt down usages when/if we change a URL. 

Can be delegated to backlog issue however